plugins/navigate.lua

```diff
@@ -0,0 +1,147 @@
+-- mod-version:1 -- lite-xl 1.16
+
+local core = require "core"
+local common = require "core.common"
+local command = require "core.command"
+local config = require "core.config"
+local keymap = require "core.keymap"
+local DocView = require "core.docview"
+
+local navigate = {
+  list = {},
+  current = nil,
+  index = 0
+}
+
+--
+-- Private functions
+--
+local function get_active_view()
+  if getmetatable(core.active_view) == DocView then
+    return core.active_view
+  end
+  return nil
+end
+
+local function add(doc)
+  -- Make new navigation point last in list
+  if navigate.index > 0 and navigate.index < #navigate.list then
+    local list_len = #navigate.list
+    for index=navigate.index+1, list_len, 1 do
+      if navigate.list[index] then
+        table.remove(navigate.list, index)
+      end
+    end
+  end
+
+  local line, col = doc:get_selection()
+  table.insert(navigate.list, {
+    filename = doc.filename,
+    line = line,
+    col = col
+  })
+
+  navigate.current = navigate.list[#navigate.list]
+  navigate.index = #navigate.list
+end
+
+local function open_doc(doc)
+  core.root_view:open_doc(
+    core.open_doc(
+      common.home_expand(
+        doc.filename
+      )
+    )
+  )
+
+  local av_doc = get_active_view().doc
+  local line, col = av_doc:get_selection()
+  if doc.line ~= line or doc.col ~= col then
+    av_doc:set_selection(doc.line, doc.col, doc.line, doc.col)
+  end
+end
+
+--
+-- Public functions
+--
+function navigate.next()
+  if navigate.index < #navigate.list then
+    navigate.index = navigate.index + 1
+    navigate.current = navigate.list[navigate.index]
+    open_doc(navigate.current)
+  end
+end
+
+function navigate.prev()
+  if navigate.index > 1 then
+    navigate.index = navigate.index - 1
+    navigate.current = navigate.list[navigate.index]
+    open_doc(navigate.current)
+  end
+end
+
+--
+-- Thread
+--
+core.add_thread(function()
+  while true do
+    local av = get_active_view()
+    if av and av.doc and av.doc.filename then
+      local doc = av.doc
+      local line, col = doc:get_selection()
+      local current = navigate.current
+      if
+        not current
+        or
+        current.filename ~= doc.filename
+        or
+        current.line ~= line
+      then
+        add(doc)
+      else
+        current.col = col
+      end
+    end
+
+    if system.window_has_focus() then
+      coroutine.yield(0.5)
+    else
+      coroutine.yield(config.project_scan_rate)
```

Comment:
That's a detail but I don't see the point of doing this. In practice when the application isn't focused no thread is run at all.

Reply:
Ahhh, I didn't know that, I did that because when I was playing with the lsp plugin some cpu usage was decreased on idle with what I think was that change. I guess it was my imagination. Will eliminate that when I get back. TY